src/test/java/uk/gov/pay/directdebit/payments/services/GovUkPayEventServiceTest.java

```diff
@@ -5,152 +5,157 @@
+        verify(govUkPayEventDao).insert(eventCaptor.capture());
+
+        GovUkPayEvent insertedEvent = eventCaptor.getValue();
+        assertThat(insertedEvent.getEventType(), is(eventType));
+        assertThat(insertedEvent.getResourceType(), is(MANDATE));
+        assertThat(insertedEvent.getMandateId(), is(Optional.of(mandateId)));
+        assertThat(insertedEvent.getPaymentId(), is(Optional.empty()));
+        assertThat(insertedEvent.getEventDate(), is(notNullValue()));
     }
 
     @Test
     public void insertMandateEvent_shouldThrowForInvalidEventTransition() {
         var previousEvent = aGovUkPayEventFixture()
-                .withResourceType(GovUkPayEvent.ResourceType.MANDATE)
+                .withResourceType(MANDATE)
                 .withMandateId(mandateId)
                 .withEventType(MANDATE_EXPIRED_BY_SYSTEM)
                 .toEntity();
-        var newEvent = aGovUkPayEventFixture()
-                .withResourceType(GovUkPayEvent.ResourceType.MANDATE)
-                .withMandateId(mandateId)
-                .withEventType(MANDATE_CREATED)
-                .toEntity();
-        
+        GovUkPayEvent.GovUkPayEventType newEventType = MANDATE_CREATED;
+
         when(govUkPayEventDao.findLatestEventForMandate(mandateId)).thenReturn(Optional.of(previousEvent));
-        when(govUkPayEventStateGraph.isValidTransition(previousEvent.getEventType(), newEvent.getEventType())).thenReturn(false);
-        
+        when(govUkPayEventStateGraph.isValidTransition(previousEvent.getEventType(), newEventType)).thenReturn(false);
+
         expectedException.expect(InvalidGovUkPayEventInsertionException.class);
         expectedException.expectMessage("GOV.UK Pay event MANDATE_CREATED is invalid following event MANDATE_EXPIRED_BY_SYSTEM");
-        
-        govUkPayEventService.storeEvent(newEvent);
+
+        govUkPayEventService.storeEventForMandate(mandate, newEventType);
     }
 
     @Test
     public void insertMandateEvent_shouldThrowForInvalidInitialEvent() {
-        var newEvent = aGovUkPayEventFixture()
-                .withResourceType(GovUkPayEvent.ResourceType.MANDATE)
-                .withMandateId(mandateId)
-                .withEventType(MANDATE_SUBMITTED)
-                .toEntity();
+        GovUkPayEvent.GovUkPayEventType eventType = MANDATE_SUBMITTED;
 
         when(govUkPayEventDao.findLatestEventForMandate(mandateId)).thenReturn(Optional.empty());
-        when(govUkPayEventStateGraph.isValidStartValue(newEvent.getEventType())).thenReturn(false);
+        when(govUkPayEventStateGraph.isValidStartValue(eventType)).thenReturn(false);
 
         expectedException.expect(InvalidGovUkPayEventInsertionException.class);
         expectedException.expectMessage("GOV.UK Pay event MANDATE_SUBMITTED is invalid when there are no previous events");
 
-        govUkPayEventService.storeEvent(newEvent);
-    }
-
-    @Test
-    public void insertMandateEvent_shouldThrowForMissingMandateId() {
```

Comment:
Why did this go?

Reply:
I changed the constructors for GovUkPayEvent, so it's no longer possible to have a null mandate id